timesketch/ui/js/controllers.js

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+'use strict';
+
+var timesketch = angular.module('timesketch', ['timesketch.directives']);
+
+// config
+timesketch.config(function($httpProvider) {
+    $httpProvider.interceptors.push(function($q, $rootScope) {
+        return {
+            'request': function(config) {
+                $rootScope.$broadcast('httpreq-start');
+                return config || $q.when(config);
+            },
+            'response': function(response) {
+                $rootScope.$broadcast('httpreq-complete');
+                return response || $q.when(response);
+            }
+        };
+    });
+    var csrftoken = document.getElementsByTagName('meta')['csrf-token'].getAttribute('content');
+    $httpProvider.defaults.headers.common['X-CSRFToken'] = csrftoken;
+});
+
+timesketch.controller('ExploreCtrl', function($scope, $http) {
+        $scope.init = function(sketch_id, view_id, timelines) {
+            $scope.sketch_id = sketch_id;
+            $scope.view_id = view_id;
+            $scope.timelines = timelines;
+            $scope.star = false;
+            $scope.filter = {};
+            $scope.filter.indices = $scope.timelines.split(",");
+            $http.get("/api/v1/sketches/" + $scope.sketch_id + "/views/" + $scope.view_id + "/").success(function(data) {
```

Comment:
I'd suggest implementing a service to work with Timesketch API. Then you can pas the service as a dependency with controller's arguments (i.e. something like function($scope, $http, timesketchApiService)). Having a dedicated service provides much better documentation and makes the code more testable. I.e. here it would look like:
timesketchService.getView($scope.sketch_id, $scope.view_id)).success(...)

Then in the tests you can just mock getView function and not care about the details on how the URL is built.